### PR TITLE
[NTV-288] Only show environmental commitments if they return from API

### PR DESCRIPTION
**Pull request by singhhari**
# 📲 What + 🤔 Why

This PR will ensure that the environmental commitments tab of the new project page only renders if values return from the API. We're performing a check in the `ProjectPageViewModel` and the `ProjectNavigationSelectorViewModel` when calling `configure` to see if `environmentalCommitments` are empty or not. I've also included some logic that makes the scrolling UX a bit more refined. In particular we don't need scrolling if the frame of the view is already showing all the buttons.

# 🛠 How

We're passing the `Project` object around and checking the `extendedProjectProperties` for an empty collection of `environmentalCommitments`. The view models will then conditionally return the appropriate set of `NavigationSection` cases which the project navigation selector view will use when laying out its arranged subviews.

# ✅ Acceptance criteria

- [x] Test a project with Environmental Commitments – https://staging.kickstarter.com/projects/getpivo/pivo-pod-x-your-pocket-sized-cameraman
- [x] Test a project without Environmental Commitments – https://staging.kickstarter.com/projects/togetherband/yawa-bands-acai-bracelets-to-preserve-the-amazon-rainforest
- [x] Verify that the correct number of buttons are rendering in the project navigation selector view, in both situations.
- [x] Verify that there is no regression in the scrolling behavior

### CODE REVIEW

**Comment by nativeksr** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Big PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Kickstarter.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -8418,6 +8418,8 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = "app-icon-alpha";
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = "Kickstarter-iOS/Alpha.entitlements";
+				CODE_SIGN_IDENTITY = "iPhone Distribution";
+				CODE_SIGN_STYLE = Manual;
```
[the remark this answers is not in the document]
Comment (msadoon) [reply Kickstarter.xcodeproj/project.pbxproj]: Similiar to this [comment](https://github.com/kickstarter/ios-oss/pull/1626#discussion_r737757755)

File: Kickstarter-iOS/Views/Controllers/ProjectPageViewControllerTests.swift [hunk 2]
```diff
@@ -234,6 +302,40 @@ internal final class ProjectPageViewControllerTests: TestCase {
...
+        parent.view.frame.size.height = device == .pad ? 1_200 : parent.view.frame.size.height
+
+        scheduler.run()
+
+        FBSnapshotVerifyView(vc.view, identifier: "lang_\(language)_device_\(device)")
+      }
+    }
+  }
+
+  func testLoggedOut_NonBacker_LiveProject_EmptyProjectProperties() {
```
Comment (msadoon) [inline Kickstarter-iOS/Views/Controllers/ProjectPageViewControllerTests.swift:332]: I think when we fix these tests to render content, we'll not use a single "all is empty" `extendedProjectProperties`, but instead go with each tab being populated and empty (2 test cases per tab).
Imo we can rename this test `testLoggedOut_NonBacker_LiveProject_EmptyProjectProperties` and `testLoggedIn_Backer_LiveProject_EmptyProjectProperties` by appending `_renderedEnvironmentalCommitments_Success`

File: Library/ViewModels/ProjectPageViewModelTests.swift [hunk 3]
```diff
@@ -249,13 +249,38 @@ final class ProjectPageViewModelTests: TestCase {
...
+      self.configureProjectNavigationSelectorView.assertDidNotEmitValue()
+
+      self.vm.inputs.viewDidLoad()
+
+      self.configureProjectNavigationSelectorView.assertDidEmitValue()
+    }
+  }
 
-    self.configureProjectNavigationSelectorView.assertDidEmitValue()
+  func testConfigureProjectNavigationSelectorView_ExtendedProjectPropertiesNil() {
```
Comment (msadoon) [inline Library/ViewModels/ProjectPageViewModelTests.swift:269]: Great test!

**Review by msadoon — CHANGES_REQUESTED** [review]
Good stuff,
almost good to go, along with the code changes, found a few UI glitches:
 Notice the delay in the nav selector appearing? Might be related to the pledge selector appearing at the same time, so are those tied together in some way?

Also, can you fix the spacing on the FAQS, technically unrelated but it would be nice to wrap those changes in this pr quickly. Notice on iPad, pretty sure its fine on iOS.



Might a regression from the merge because I thought we fixed this already.

File: Kickstarter.xcodeproj/project.pbxproj (hunk 1 above)
[the remark this answers is not in the document]
Comment (singhhari) [reply Kickstarter.xcodeproj/project.pbxproj]: i don't actually see this on the diff anymore so I think after I updated the feature branch this resolved itself.

[the remark this answers is not in the document]
Comment (singhhari) [reply Kickstarter.xcodeproj/project.pbxproj]: Says Outdated next to the file name above too

**Comment by singhhari** [comment]
> Good stuff, almost good to go, along with the code changes, found a few UI glitches:  Notice the delay in the nav selector appearing? Might be related to the pledge selector appearing at the same time, so are those tied together in some way?
> 
> Also, can you fix the spacing on the FAQS, technically unrelated but it would be nice to wrap those changes in this pr quickly. Notice on iPad, pretty sure its fine on iOS.
> 
> 
> 
> Might a regression from the merge because I thought we fixed this already.

The issue with the navigation selector is a tricky one because this PR officially makes it dependent on the network requests response. The logic im applying is actually pretty straightforward: take the project from the response, map for the extendedProjectProperties and emit that. For whatever reason, when applying breakpoints, the configuration of the childViewController occurs before the configuration of the project navigation selector or pledgeCTAContainerView despite all three awaiting the response. Not sure how we should handle this but this does kinda make sense since we are introducing a dependency of the view to the API.

**Review by msadoon — APPROVED** [review]
good to go, just create a new ticket for the delayed nav selector if cannot in this pr.

**Comment by singhhari** [comment]
> good to go, just create a new ticket for the delayed nav selector if cannot in this pr.

Ended up creating a new ticket for this since I'm going to be out for two weeks and do not want to block this PR from merging.

**Pull request merged**